presto-main/src/main/java/com/facebook/presto/execution/scheduler/NodeMap.java

```diff
@@ -67,4 +67,9 @@ public NodeMap(
     {
         return coordinatorNodeIds;
     }
+
+    public Set<HostAddress> getHostAndAddress()
+    {
+        return nodesByHostAndPort.keySet();
+    }
```

Comment:
Instead of having this, create the following function and remove class `NodeSelectionUtils`
```java
    NodeProvider getNodeProvider(NodeSelectionHashFunction nodeSelectionHashFunction)
    {
        switch (nodeSelectionHashFunction) {
            case MODULAR:
                new ModularHashingNodeProvider<>(sortedNodes(this));
            case CONSISTENT_HASHING:
                new ConsistentHashingNodeProvider<>(nodesByHostAndPort.keySet());
            default:
                throw ...
        }
    }

    private static List<HostAddress> sortedNodes(NodeMap nodeMap)
    {
        return nodeMap.getNodesByHostAndPort().values().stream()
                .sorted(comparing(InternalNode::getNodeIdentifier)).map(InternalNode::getHostAndPort)
                .collect(toImmutableList());
    }
```

Reply:
Will create the `getNodeProvider`

but removing `NodeSelectionUtils`, I think we just moved to `NodeSelectionUtils` according to the comments last time
https://github.com/prestodb/presto/pull/13966#discussion_r378665002
Also `TopologyAwareNodeSelector` is also using it, so I think it's reasonable to keep it there